packages/jaeger-ui/src/api/jaeger.js

```diff
@@ -48,7 +48,19 @@ function getJSON(url, options = {}) {
 
   return fetch(`${url}${queryStr}`, init).then(response => {
     if (response.status < 400) {
-      return response.json();
+      return response.json().then(body => {
+        if (body && Array.isArray(body.errors) && body.errors.length > 0) {
```

Comment:
A function called `getJSON` cannot assume to know the structure of the response. Why do we need to throw exception in this lower level function instead of handling the error where needed?